pkg/kfuzztest/kfuzztest.go

```diff
@@ -0,0 +1,190 @@
+// Package kfuzztest exposes functions discovering KFuzzTest test cases from a
+// vmlinux binary and parsing them into syzkaller-compatible formats.
+// The general flow includes:
+//   - Creating an Extractor that extracts these test cases from the binary
+//   - Creating a Builder that takes the extractor's output and returns some
+//     compatible encoding of the test cases that were discovered
+package kfuzztest
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/google/syzkaller/pkg/ast"
+	"github.com/google/syzkaller/pkg/compiler"
+	"github.com/google/syzkaller/pkg/kcov"
+	"github.com/google/syzkaller/prog"
+	"github.com/google/syzkaller/sys/targets"
+)
+
+type SyzField struct {
```

Comment:
I don't know how much of that can be reused (or whether we should strive to do it in this case), but here's one more place in syzkaller where we define similar structures:

https://github.com/google/syzkaller/blob/master/pkg/declextract/entity.go